### PR TITLE
[Rebranch] Fix SILOptimizer tests

### PR DESCRIPTION
**Pull request by etcwilde**
Went through and removed the undefined variables in the tests in the SILOptimizer tests since FileCheck complains about those now.

Please check to ensure that I've done this right and that you don't want me to take a different approach to fixing this.

rdar://82050671

### CODE REVIEW

**Review by meg-gupta — APPROVED** [review]
thanks! lgtm

File: test/SILOptimizer/array_element_propagation.sil [hunk 1]
```diff
@@ -222,7 +222,6 @@ sil @unknown_use : $@convention(thin) () -> () {
 // CHECK:      [[ASFUN:%.*]] = function_ref @arrayAdoptStorage
 // CHECK-NEXT: [[ARR:%.*]] = apply [[ASFUN]]
 // CHECK-NEXT: [[OWNER:%.*]] = tuple_extract [[ARR]]{{.*}}, 0
-// CHECK-NOT:    apply [[ACFUN]]
```
Comment (eeckstein) [inline test/SILOptimizer/array_element_propagation.sil:225]: The `CHECK-NOT` lines are actually essential for all the tests. They check if the optimization could really remove those instructions. I know, they didn't have an effect so far, but it should be easy to fix those, e.g. by removing the argument lit-variable. E.g.
```
// CHECK-NOT:    apply
```

**Review by eeckstein — COMMENTED** [review]
Can you try to keep the ``CHECK-NOT`` lines by just removing the lit-variables in the pattern?

File: test/SILOptimizer/array_element_propagation.sil (hunk 1 above)
Comment (etcwilde) [reply test/SILOptimizer/array_element_propagation.sil:225]: There are other applies between the preceding CHECK-NEXT and the following CHECK, so the test fails.
Specifically the application of the arrayInit and the array meta type.

Comment (etcwilde) [reply test/SILOptimizer/array_element_propagation.sil:225]: The current output between the arrayAdoptStorage and arrayAppendContentsOf is
```sil
  // function_ref arrayAdoptStorage
  %5 = function_ref @arrayAdoptStorage : $@convention(thin) (@owned AnyObject, MyInt, @thin Array<MyInt>.Type) -> @owned (Array<MyInt>, UnsafeMutablePointer<MyInt>) // user: %6
  %6 = apply %5(%3, %2, %4) : $@convention(thin) (@owned AnyObject, MyInt, @thin Array<MyInt>.Type) -> @owned (Array<MyInt>, UnsafeMutablePointer<MyInt>) // users: %8, %7
  %7 = tuple_extract %6 : $(Array<MyInt>, UnsafeMutablePointer<MyInt>), 0 // user: %25
  %8 = tuple_extract %6 : $(Array<MyInt>, UnsafeMutablePointer<MyInt>), 1 // user: %9
  %9 = struct_extract %8 : $UnsafeMutablePointer<MyInt>, #UnsafeMutablePointer._rawValue // user: %10
  %10 = pointer_to_address %9 : $Builtin.RawPointer to [strict] $*MyInt // user: %13
  %11 = integer_literal $Builtin.Int64, 27        // user: %12
  %12 = struct $MyInt (%11 : $Builtin.Int64)      // users: %22, %13
  store %12 to %10 : $*MyInt                      // id: %13
  %14 = alloc_stack $Array<MyInt>                 // users: %23, %26, %18
  %15 = metatype $@thin Array<MyInt>.Type         // user: %17
  // function_ref arrayInit
  %16 = function_ref @arrayInit : $@convention(method) (@thin Array<MyInt>.Type) -> @owned Array<MyInt> // user: %17
  %17 = apply %16(%15) : $@convention(method) (@thin Array<MyInt>.Type) -> @owned Array<MyInt> // user: %18
  store %17 to %14 : $*Array<MyInt>               // id: %18
  // function_ref arrayAppendContentsOf
  %19 = function_ref @arrayAppendContentsOf : $@convention(method) (@owned Array<MyInt>, @inout Array<MyInt>) -> ()
  // function_ref Array.append(_:)
  %20 = function_ref @$sSa6appendyyxnF : $@convention(method) <τ_0_0> (@in τ_0_0, @inout Array<τ_0_0>) -> () // user: %23
```
We are specifically getting hung up on the application of `arrayInit`.

Comment (eeckstein) [reply test/SILOptimizer/array_element_propagation.sil:225]: ok, then there should be a
```
[[ACFUNC:%.*]] = function_ref @arrayAppendContentsOf
```
before the `CHECK-NOT: apply [[ACFUNC]]`.
The point here is that this function is not called.

Comment (etcwilde) [reply test/SILOptimizer/array_element_propagation.sil:225]: Done

File: test/SILOptimizer/mem2reg.sil [hunk 2]
```diff
@@ -298,7 +298,6 @@ bb2:
 
 // CHECK: bb3([[ARG:%.*]] : $@callee_owned () -> Int):
 bb3:
-// CHECK-NOT: load [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg.sil:301]: Can you fix the other CHECK-NOT lines, too?

Comment (etcwilde) [reply test/SILOptimizer/mem2reg.sil:301]: I don't actually know what this is supposed to be resolved to. I see that there is a stack defined above in the `CHECK-NOT`, but that doesn't seem to define variables. Can you provide a patch of the changes you want? Thanks.

Comment (eeckstein) [reply test/SILOptimizer/mem2reg.sil:301]: This should be
```
// CHECK-NOT: load
```

File: test/SILOptimizer/mem2reg.sil [hunk 3]
```diff
@@ -311,10 +310,8 @@ bb3:
   //       block.
 // CHECK: bb4
 bb4:
-// CHECK-NOT: destroy_addr [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg.sil:314]:
```
// CHECK-NOT: destroy_addr
```

File: test/SILOptimizer/mem2reg.sil [hunk 4]
```diff
@@ -311,10 +310,8 @@ bb3:
...
 // CHECK: strong_release [[ARG]]
   destroy_addr %1 : $*@callee_owned () -> Int
-// CHECK-NOT: dealloc_stack [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg.sil:317]:
```
// CHECK-NOT: dealloc_stack
```

File: test/SILOptimizer/mem2reg_ossa.sil [hunk 5]
```diff
@@ -299,7 +299,6 @@ bb2:
 
 // CHECK: bb3([[ARG:%.*]] : @owned $@callee_owned () -> Int):
 bb3:
-// CHECK-NOT: load [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg_ossa.sil:302]:
```
// CHECK-NOT: load
```

File: test/SILOptimizer/mem2reg_ossa.sil [hunk 6]
```diff
@@ -311,10 +310,8 @@ bb3:
   //       block.
 // CHECK: bb4
 bb4:
-// CHECK-NOT: destroy_addr [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg_ossa.sil:314]:
```
// CHECK-NOT: destroy_addr
```

File: test/SILOptimizer/mem2reg_ossa.sil [hunk 7]
```diff
@@ -311,10 +310,8 @@ bb3:
...
 // CHECK: destroy_value [[ARG]]
   destroy_addr %1 : $*@callee_owned () -> Int
-// CHECK-NOT: dealloc_stack [[STACK]]
```
Comment (eeckstein) [inline test/SILOptimizer/mem2reg_ossa.sil:317]:
```
// CHECK-NOT: dealloc_stack
```

File: test/SILOptimizer/super_init.swift [hunk 8]
```diff
@@ -7,7 +7,6 @@
 // CHECK:         [[NEW_SELF:%.*]] = apply [[SUPER_INIT]]
 
 // CHECK-LABEL: sil hidden [noinline] @$s10super_init3BarC{{[_0-9a-zA-Z]*}}fc
-// CHECK-NOT:     super_method [[ORIG_SELF]] : $Bar, #Foo.init!initializer
```
Comment (eeckstein) [inline test/SILOptimizer/super_init.swift:10]:
```
// CHECK-NOT:     super_method
```

**Review by eeckstein — APPROVED** [review]
LGTM, thanks!

**Pull request merged**